deploy.ps1

```diff
@@ -0,0 +1,165 @@
+$ErrorActionPreference = "Stop"
+
+$RepoRoot = $PSScriptRoot
+$OutputDir = "$PSScriptRoot/out"
+
+# TODO: Set these better
+$tf_share = "zachterraformstorage"
+$kv_name = "mics-kv"
+$acr_name = "mics233.azurecr.io"
+
+Import-Module -Name "./modules/jaw"
+
+"Requirements" | % {
+    if (-not (Get-InstalledModule  $_)) {
+        Install-Module $_ -Force
+    }
+    Import-Module $_
+}
+
+# Auth Azure and gather subscription secrets
+$azureReqs = @(
+    @{
+        Name     = "Azure Login"
+        Describe = "Authenticate Azure Session"
+        Test     = { [Boolean] (az account show) }
+        Set      = { az login }
+    },
+    @{  # This could be done idempotently with a test,
+        # but refreshing the secrets every run allows for
+        # new secrets to be added easily
+        Name     = "Keyvault Secrets"
+        Describe = "Inject Secrets into Session"
+        Set      = {
+            $KEYVAULTNAME = $kv_name
+            $SECRETS = ( $(az keyvault secret list --vault-name $KEYVAULTNAME | jq '.[].id' -r | sed 's/.*\/\([^/]\+\)$/\1/') )
+            $SECRETS | % {
+                $SECRET = $(az keyvault secret show --name $_ --vault-name $KEYVAULTNAME | jq '.value' -r)
+                $NAME = $_.Replace("-", "_")
+                [Environment]::SetEnvironmentVariable($NAME, $SECRET)
+            }
+        }
+    }
+)
+
+# Provision Infra
+$tfReqs = @(
+    @{
+        Name     = "Terraform init"
+        Describe = "Initialize terraform environment"
+        Test     = { Test-Path "$PSScriptRoot/tf/.terraform" }
+        Set      = {
+            Set-Location -Path "tf"
+            terraform init -backend-config="storage_account_name=$($tf_share)" `
+                -backend-config="container_name=tfstate" `
+                -backend-config="access_key=$($env:terraform_storage_key)" `
+                -backend-config="key=mics.tfstate"
+        }
+    },
+    @{
+        Name     = "Terraform plan"
+        Describe = "Plan terraform environment"
+        Test     = { Test-Path "$OutputDir/out.plan" }
+        Set      = {
+            New-Item -Path "$OutputDir" -ItemType Directory -Force
+            terraform plan -out "$OutputDir/out.plan"
+        }
+    },
+    @{
+        Name     = "Terraform Apply"
+        Describe = "Apply Terraform plan"
+        Test     = { Test-Path "$OutputDir/azurek8s" }
+        Set      = {
+            terraform apply "$OutputDir/out.plan" | Write-Information
+            terraform output kube_config | Out-File "$OutputDir/azurek8s"
+            Set-Location $RepoRoot
+        }
+    }
+)
+
+# Docker cooking
+$dockerReqs = @(
+    @{
+        Name     = "Find Docker Services"
+        Describe = "Enumerate Containers"
+        Set      = {
+            Set-k8sConfig -AppPath "./app" -OutPath "./out"
```

Comment:
This parses /app in search of Dockerfiles, parses them, and then writes a JSON structure with information about where it's located, what commands it runs, and inferred names of services it finds.

Reply:
Well done